README.md

```diff
@@ -16,6 +16,10 @@
 
 - [Design goals](#design-goals)
 - [Integration](#integration)
+  - [CMake](#cmake)
+    - [External](#external)
```

Comment:
The table of contents seems to be rather crowded.

- Please leave "External", "Embedded", and "Supporting Both" out of the table of contents.
- Please add "Package Managers" to the table of contents.